### PR TITLE
Adds Custom Properties

### PR DESCRIPTION
**Pull request by bgustafson**
Adds custom properties on tests. This is behind a feature flag still so I will remove from draft status once we get products go ahead to release.

### CODE REVIEW

**Comment by bgustafson** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by bgustafson** [comment]
recheck

File: synthetics/resource_api_check_v2.go [hunk 1]
```diff
@@ -194,6 +194,23 @@ func resourceApiCheckV2() *schema.Resource {
 							Default:      "round_robin",
 							ValidateFunc: validation.StringMatch(regexp.MustCompile(`(^concurrent$|^round_robin$)`), "Setting must match concurrent or round_robin"),
 						},
+						"custom_properties": {
```
Comment (greatestusername-splunk) [inline synthetics/resource_api_check_v2.go:197]: I think you'll want to add a `ValidateFunc` for `custom_properties` in each of the schemas. Was fiddling and realized that adding a key/value with a whitespace (or other unacceptable character) will error out with a 422.

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.18
 require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.24.1
 	github.com/splunk/syntheticsclient v1.0.3
-	github.com/splunk/syntheticsclient/v2 v2.0.4
+	github.com/splunk/syntheticsclient/v2 v2.0.5
```
Comment (greatestusername-splunk) [inline go.mod:8]: mind doing a `go mod vendor` also to update any required `vendor` files (probably just `modules.txt`)?

**Comment by greatestusername-splunk** [comment]
I merged https://github.com/splunk/terraform-provider-synthetics/pull/31
So you should be able to pull in the new acceptance tests and add `custom_properties` to those tests. 

Thanks!

File: go.mod (hunk 2 above)
Comment (bgustafson) [reply go.mod:8]: done

File: synthetics/resource_api_check_v2.go (hunk 1 above)
Comment (bgustafson) [reply synthetics/resource_api_check_v2.go:196]: Added regex to match backend validations

Comment (greatestusername-splunk) [reply synthetics/resource_api_check_v2.go:196]: Thank you!

**Review by greatestusername-splunk — APPROVED** [review]
LGTM!
Thanks for your contributions!

**Pull request merged**